### PR TITLE
Restrict Julia version in Project.toml

### PR DESCRIPTION
**Pull request by torfjelde**
Do we actually support Julia 1.4 and 1.5? Or is it only Julia 1.3 and 1.6?

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -59,4 +59,4 @@ StatsBase = "0.32, 0.33"
 StatsFuns = "0.8, 0.9"
 Tracker = "0.2.3"
 ZygoteRules = "0.2"
-julia = "1.3"
+julia = "1.3, 1.4, 1.5, 1.6"
```
Comment (rikhuijzer) [inline Project.toml:62]:
```suggestion
julia = "1.6"
```

**Comment by rikhuijzer** [comment]
Usually, Julia developers run tests for some lower bound and upper bound version and if those pass assume that everything in between also works.

**Comment by devmotion** [comment]
AFAIK we supported Julia >= 1.3, including 1.4 and 1.5, we just only tested with the oldest compatible version and the stable Julia version only to reduce CI.

Even with this PR users with Julia 1.7 will be able to install older releases of Turing. If we want to avoid this we would have to change the compat entry for older versions in the registry directly.

File: Project.toml (hunk 1 above)
Comment (devmotion) [reply Project.toml:62]: We don't have to be more restrictive than necessary 🤷‍♂️ We can easily drop support for < 1.6 when it requires special effort to support older versions. As long as it just works, I don't think we have to remove them.

**Review by devmotion — APPROVED** [review]

**Comment by rikhuijzer** [comment]
The package [RetroCap.jl](https://github.com/JuliaRegistries/RetroCap.jl) might be useful.

**Pull request merged**